Index: lm-sdk-ide/dist/qt5/qtbase/src/3rdparty/icu_dependency.pri
===================================================================
--- lm-sdk-ide.orig/dist/qt5/qtbase/src/3rdparty/icu_dependency.pri
+++ /dev/null
@@ -1,13 +0,0 @@
-win32 {
-    CONFIG(static, static|shared) {
-        CONFIG(debug, debug|release) {
-            LIBS_PRIVATE += -lsicuind -lsicuucd -lsicudtd
-        } else {
-            LIBS_PRIVATE += -lsicuin -lsicuuc -lsicudt
-        }
-    } else {
-        LIBS_PRIVATE += -licuin -licuuc -licudt
-    }
-} else {
-    LIBS_PRIVATE += -licui18n -licuuc -licudata
-}
Index: lm-sdk-ide/dist/qt5/qtbase/src/corelib/configure.json
===================================================================
--- lm-sdk-ide.orig/dist/qt5/qtbase/src/corelib/configure.json
+++ lm-sdk-ide/dist/qt5/qtbase/src/corelib/configure.json
@@ -44,7 +44,6 @@
         },
         "icu": {
             "label": "ICU",
-            "export": "",
             "test": "unix/icu",
             "sources": [
                 {
Index: lm-sdk-ide/dist/qt5/qtbase/src/corelib/tools/tools.pri
===================================================================
--- lm-sdk-ide.orig/dist/qt5/qtbase/src/corelib/tools/tools.pri
+++ lm-sdk-ide/dist/qt5/qtbase/src/corelib/tools/tools.pri
@@ -138,7 +138,7 @@ qtConfig(system-zlib) {
 }
 
 qtConfig(icu) {
-    include($$PWD/../../3rdparty/icu_dependency.pri)
+    QMAKE_USE_PRIVATE += icu
 
     SOURCES += tools/qlocale_icu.cpp \
                tools/qcollator_icu.cpp
